sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -0,0 +1,232 @@
+        return self.__dict__[key]
+
+    def __contains__(self, key):
+        return key in self.__dict__
+
+    def __repr__(self):
+        return str(self)
+
+    def __len__(self):
+        return len(self.keys())
+
+    def __delitem__(self, key):
+        self.__dict__[key] = None
+
+    def __eq__(self, other):
+        """Compare objects by comparing all attributes."""
+        if isinstance(other, self.__class__):
+            return self.__dict__ == other.__dict__
+        return False
+
+    def __ne__(self, other):
+        """Compare objects by comparing all attributes."""
+        return not self.__eq__(other)
+
+    def __str__(self):
+        return str({k: v for k, v in self.__dict__.items() if not k.startswith('_')})
+
+    def has_key(self, k, **kwargs):
+        return k in self.__dict__
+
+    def update(self, *args, **kwargs):
+        return self.__dict__.update(*args, **kwargs)
+
+    def keys(self, **kwargs):
+        return [k for k in self.__dict__ if not k.startswith('_')]
+
+    def values(self, **kwargs):
+        return [v for k, v in self.__dict__.items() if not k.startswith('_')]
+
+    def items(self, **kwargs):
+        return [(k, v) for k, v in self.__dict__.items() if not k.startswith('_')]
+
+    def get(self, key, default=None, **kwargs):
+        if key in self.__dict__:
+            return self.__dict__[key]
+        return default
+
+
+class DeserializedEvent(DictMixin):
+    """The container for the deserialized event model and mapping of event envelope properties.
+        :param dict args: dict
+    """
+    # class variable
+    #_event_type_mappings = {'Microsoft.Storage.BlobCreated': StorageBlobCreatedEventData()}
+
+    def __init__(self, *args, **kwargs):
+        # type: (Any) -> None
+        self._update(*args, **kwargs)
+        self._model = None
+
+    def _update(self, *args, **kwargs):
+        for k, v in dict(*args, **kwargs).items():
+            self[k] = v
+
+    @property
+    def model(self):
+        # type: () -> Union[CloudEvent, EventGridEvent]
+        """
+        Returns strongly typed EventGridEvent/CloudEvent object defined by the format of the properties.
+        All properties of the model are strongly typed (ie. for an EventGridEvent, event_time property will return a datetime.datetime object).
+
+        model.data: Returns a system event type(StorageBlobCreated, StorageBlobDeleted, etc.). If model.type/model.event_type is not defined in the 
+          system registry, returns None.
+
+        :rtype: Union[CloudEvent, EventGridEvent]
+        """
+        if not self._model:
+            if 'specversion' in self.keys():
+                self._model = CloudEvent(**self)
+                self._model.time = dt.datetime.strptime(self._model.time, "%Y-%m-%dT%H:%M:%S.%fZ")
+                # replace all below, only for demo
+                if self['type'] == "Microsoft.Storage.BlobCreated":
+                    self._model.data = self._model.data.replace("'", "\"")
```

Comment:
Suspicious? Can I get a rationale?

Reply:
Originally, the _model.data was a string. So, when I tried to create a dict using json.loads, the string event had ' instead of " and null instead of None. This resulted in an error, that I tried to fix by replacing both ' and None. Have already changed this part to use Model.deserialize.